pkg/controller/llmisvc/router.go

```diff
@@ -0,0 +1,195 @@
+		for _, ref := range llmSvc.Spec.Router.Gateway.Refs {
+			httpRoute.Spec.CommonRouteSpec.ParentRefs = append(httpRoute.Spec.CommonRouteSpec.ParentRefs, gatewayapi.ParentReference{
+				// TODO(api): With this structure we are missing the ability to narrow a section of targeted gateway by the route we are creating
+				// missing SectionName and Port
+				Name:      ref.Name,
+				Namespace: &ref.Namespace,
+				Group:     ptr.To(gatewayapi.Group("gateway.networking.k8s.io")),
+				Kind:      ptr.To(gatewayapi.Kind("Gateway")),
+			})
+		}
+	}
+
+	return httpRoute
+}
+
+func RouterLabels(llmSvc *v1alpha1.LLMInferenceService) map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/component": "llminferenceservice-router",
+		"app.kubernetes.io/name":      llmSvc.GetName(),
+		"app.kubernetes.io/part-of":   "llminferenceservice",
+	}
+}
+
+func (r *LLMInferenceServiceReconciler) reconcileHTTPRoute(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService, expected *gatewayapi.HTTPRoute) error {
+	curr := &gatewayapi.HTTPRoute{}
+
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(expected), curr)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get HTTPRoute %s/%s: %w", expected.GetNamespace(), expected.GetName(), err)
+	}
+	if apierrors.IsNotFound(err) {
+		return r.createHTTPRoute(ctx, llmSvc, expected)
+	}
+
+	if equality.Semantic.DeepDerivative(expected.Spec, curr.Spec) &&
+		equality.Semantic.DeepEqual(expected.Labels, curr.Labels) &&
+		equality.Semantic.DeepEqual(expected.Annotations, curr.Annotations) &&
+		metav1.IsControlledBy(curr, llmSvc) {
+		return nil
+	}
+	return r.updateHTTPRoute(ctx, llmSvc, curr, expected)
+}
+
+func (r *LLMInferenceServiceReconciler) createHTTPRoute(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService, httpRoute *gatewayapi.HTTPRoute) error {
+	if err := r.Client.Create(ctx, httpRoute); err != nil {
+		return fmt.Errorf("failed to create HTTPRoute %s/%s: %w", httpRoute.GetNamespace(), httpRoute.GetName(), err)
+	}
+	r.Recorder.Eventf(llmSvc, corev1.EventTypeNormal, "Created", "Created HTTPRoute %s/%s", httpRoute.GetNamespace(), httpRoute.GetName())
+
+	return nil
+}
+
+func (r *LLMInferenceServiceReconciler) updateHTTPRoute(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService, curr, httpRoute *gatewayapi.HTTPRoute) error {
+	if !metav1.IsControlledBy(curr, llmSvc) {
+		return fmt.Errorf("failed to update HTTPRoute %s/%s: it is not controlled by LLMInferenceService %s/%s",
+			httpRoute.GetNamespace(), httpRoute.GetName(),
+			llmSvc.GetNamespace(), llmSvc.GetName(),
+		)
+	}
+
+	if err := r.Client.Update(ctx, httpRoute); err != nil {
+		return fmt.Errorf("failed to update HTTPRoute %s/%s: %w", httpRoute.GetNamespace(), httpRoute.GetName(), err)
+	}
+	r.Recorder.Eventf(llmSvc, corev1.EventTypeNormal, "Updated", "Updated HTTPRoute %s/%s", httpRoute.GetNamespace(), httpRoute.GetName())
+
+	return nil
+}
```

Comment:
would it work if we use the generic `updateObject`

```
	return r.updateObject(ctx, llmSvc, curr, expected, semanticHTTPRouteIsEqual)
```

Reply:
Addressed in 2b9bf536a